src/ucp/api/ucp_def.h

```diff
@@ -385,13 +385,30 @@ typedef void (*ucp_tag_recv_callback_t)(void *request, ucs_status_t status,
  * @brief UCP worker wakeup events mask.
  *
  * The enumeration allows specifying which events are expected on wakeup, though
- * empty events are possible.
+ * empty events are possible for any type of events excepting @ref UCP_WAKEUP_TX
+ * and UCP_WAKEUP_RX which are the most common events.
+ *
+ * @note UCP doesn't generate POLLOUT-like events (see poll man pages) since
+ * it's always ready to initiate outgoing operations (e.g any type of sending,
+ * atomic or RMA). Hoverer, they can be queued due to limitation of resources.
+ * Send completions are reported by POLLIN-like events.
  */
 typedef enum ucp_wakeup_event_types {
     UCP_WAKEUP_RMA         = UCS_BIT(0), /**< Remote memory access send completion */
     UCP_WAKEUP_AMO         = UCS_BIT(1), /**< Atomic operation send completion */
     UCP_WAKEUP_TAG_SEND    = UCS_BIT(2), /**< Tag send completion  */
     UCP_WAKEUP_TAG_RECV    = UCS_BIT(3), /**< Tag receive completion */
+    UCP_WAKEUP_TX          = UCS_BIT(10),/**< This event type will generate an
+                                              event on completion of any
+                                              outgoing operation (or even its
+                                              part, according to underlying
+                                              protocol) wheter any type of
+                                              sending, atomic or RMA. */
```

Comment:
Suggested re-wording:

This event type will generate an event on completion of any outgoing operation (complete or
partial, according to the underlying protocol) for any type of transfer (send, atomic, or RMA).